### PR TITLE
Rename unique feature to globalsequence.

### PR DESCRIPTION
**Pull request by baldersheim**
Invert so that the first doc from the first node will sort first,
followed by the first doc from the second node, and the the last doc of the last node at the end.
It will produce number in the range (1 << 48) down to zero.
Sequence = (1 << 48) - ((docId << 16) | disributionKey)

@havardpe and @geirst PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+void
+GlobalSequenceBlueprint::visitDumpFeatures(const IIndexEnvironment &, IDumpFeatureVisitor &) const
+{
+}
+
+bool
+GlobalSequenceBlueprint::setup(const IIndexEnvironment & env, const ParameterList & )
+{
+    _distributionKey = env.getDistributionKey();
+    assert( _distributionKey < 0x80000);
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:47]: what about the 3 bits we allow to overlap between distribution key and docid?

File: searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    uint32_t _distributionKey;
+
+public:
+    GlobalSequenceExecutor(uint32_t distributionKey)
+        : _distributionKey(distributionKey)
+    {
+    }
+
+    void execute(uint32_t docId) override {
+        outputs().set_number(0, ((1ul << 48u) - ((uint64_t(docId) << 16u) | _distributionKey)));
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:26]: consider moving the magic to an inlined function in the header that can also be called by the test

File: searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+};
+
+}
+
+GlobalSequenceBlueprint::GlobalSequenceBlueprint() :
+    Blueprint("globalsequence"),
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:33]: consider camel case to match naming of other features

Comment (geirst) [reply searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:33]: Yes, the name should use came case.

File: searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:47]: Ouch, thanks. That should be 0x10000

File: searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp (hunk 3 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:33]: globalsequence -> globalSequence

File: searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:26]: Fixed

**Pull request merged**